core/chain_indexer.go

```diff
@@ -473,3 +486,94 @@ func (c *ChainIndexer) removeSectionHead(section uint64) {
 
 	c.indexDb.Delete(append([]byte("shead"), data[:]...))
 }
+
+func (c *ChainIndexer) addOutpointsToIndexer(addressOutpoints map[string][]*types.OutPoint, config params.ChainConfig, block *types.Block) {
+	utxos := block.QiTransactions()
+
+	for _, tx := range utxos {
+		for i, in := range tx.TxIn() {
+			outpoint := in.PreviousOutPoint
+
+			// Coinbase
+			if i == 0 {
+				continue
+			}
+			addr := crypto.PubkeyBytesToAddress(in.PubKey, config.Location).Hex()
+			outpointsForAddress := addressOutpoints[addr]
+
+			for i := 0; i < len(outpointsForAddress); i++ {
+				if outpointsForAddress[i] == &outpoint {
+					outpointsForAddress = append(outpointsForAddress[:i], outpointsForAddress[i+1:]...)
+					addressOutpoints[addr] = outpointsForAddress
+
+					break
+				}
+			}
+		}
+
+		for i, out := range tx.TxOut() {
+
+			address := out.Address
+			outpoint := types.OutPoint{
```

Comment:
Why not store the value of the utxo as well?

Reply:
Good idea

Reply:
What do you mean? The UTXO denominations are fully described in every transaction you process in the reorg loop

Reply:
My comment was on the TxOuts, which have the denominations